WordPress/Classes/ViewRelated/Post/PostListViewController.swift

```diff
@@ -0,0 +1,685 @@
+    static private let statsStoryboardName = "SiteStats"
+    static private let currentPostListStatusFilterKey = "CurrentPostListStatusFilterKey"
+    static private let currentPostAuthorFilterKey = "CurrentPostAuthorFilterKey"
+    
+    // TODO: low cap on first char!
+    
+    static private let statsCacheInterval = NSTimeInterval(300) // 5 minutes
+    static private let postCardEstimatedRowHeight = Float(100.0)
+    static private let postCardRestoreCellRowHeight = Float(54.0)
+    static private let postListHeightForFooterView = CGFloat(34.0)
+    
+    @IBOutlet private var textCellForLayout : PostCardTableViewCell!
+    @IBOutlet private var imageCellForLayout : PostCardTableViewCell!
+    @IBOutlet private weak var authorFilterSegmentedControl : UISegmentedControl!
+    
+    // MARK: Initializers & deinitializers
+    
+    deinit {
+        PrivateSiteURLProtocol.unregisterPrivateSiteURLProtocol()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        
+        PrivateSiteURLProtocol.registerPrivateSiteURLProtocol()
+    }
+    
+    // MARK: - Convenience constructors
+    
+    class func controllerWithBlog(blog: Blog) -> PostListViewController {
+        
+        let storyBoard = UIStoryboard(name: "Posts", bundle: NSBundle.mainBundle())
+        let controller = storyBoard.instantiateViewControllerWithIdentifier("PostListViewController") as! PostListViewController
+        
+        controller.blog = blog
+        controller.restorationClass = self
+
+        return controller
+    }
+    
+    // MARK: - UIViewControllerRestoration
+    
+    class func viewControllerWithRestorationIdentifierPath(identifierComponents: [AnyObject], coder: NSCoder) -> UIViewController? {
+        
+        let context = ContextManager.sharedInstance().mainContext
+        
+        guard let blogID = coder.decodeObjectForKey(postsViewControllerRestorationKey) as? String,
+            let objectURL = NSURL(string: blogID),
+            let objectID = context.persistentStoreCoordinator?.managedObjectIDForURIRepresentation(objectURL),
+            let restoredBlog = try? context.existingObjectWithID(objectID) as! Blog else {
+                
+            return nil
+        }
+        
+        return self.controllerWithBlog(restoredBlog)
+    }
+    
+    // MARK: - UIStateRestoring
+    
+    override func encodeRestorableStateWithCoder(coder: NSCoder) {
+        
+        let objectString = blog?.objectID.URIRepresentation().absoluteString
+        
+        coder.encodeObject(objectString, forKey:self.dynamicType.postsViewControllerRestorationKey)
+        
+        super.encodeRestorableStateWithCoder(coder)
+    }
+    
+    // MARK: - UIViewController
+    
+    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
+        super.postListViewController = (segue.destinationViewController as! UITableViewController)
```

Comment:
`super.postListViewController = segue.destinationViewController as? UITableViewController`  for paranoia.